From ca9434f55dd0fe8148cf63aadafdcc52de6d22b2 Mon Sep 17 00:00:00 2001
From: qianmenx <qianx.meng@intel.com>
Date: Thu, 25 Jul 2019 10:40:28 +0800
Subject: [PATCH 1496/1548] media: intel-ipu4: add header file to some file for
 build warnings

add header file to dw9714.c as3638.h ipu-isys-isa-fw.h
  ipu-psys.h

Change-Id: If4c714c4adb771452417e943de1410875cc06e25
Tracked-On: PKT-2588
Tracked-On: #H1507330328
Signed-off-by: qianmenx <qianx.meng@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/i2c/dw9714.c           | 1 +
 include/media/as3638.h               | 6 ++++++
 include/uapi/linux/ipu-isys-isa-fw.h | 6 ++++++
 include/uapi/linux/ipu-psys.h        | 4 ++++
 4 files changed, 17 insertions(+)

diff --git a/drivers/media/i2c/dw9714.c b/drivers/media/i2c/dw9714.c
index 3dc2100470a1..660b2cf0891b 100644
--- a/drivers/media/i2c/dw9714.c
+++ b/drivers/media/i2c/dw9714.c
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
 // Copyright (c) 2015--2017 Intel Corporation.
 
+#include <linux/version.h>
 #include <linux/delay.h>
 #include <linux/i2c.h>
 #include <linux/module.h>
diff --git a/include/media/as3638.h b/include/media/as3638.h
index 8e847efaac9f..6fd6b6990e73 100644
--- a/include/media/as3638.h
+++ b/include/media/as3638.h
@@ -4,6 +4,12 @@
 #ifndef __AS3638_H__
 #define __AS3638_H__
 
+#ifdef __KERNEL__
+#include <linux/types.h>
+#else
+#include <stdint.h>
+#endif
+
 #define AS3638_NAME			"as3638"
 #define AS3638_I2C_ADDR			0x30
 
diff --git a/include/uapi/linux/ipu-isys-isa-fw.h b/include/uapi/linux/ipu-isys-isa-fw.h
index f0914bce7966..3a4ee6dee14a 100644
--- a/include/uapi/linux/ipu-isys-isa-fw.h
+++ b/include/uapi/linux/ipu-isys-isa-fw.h
@@ -4,6 +4,12 @@
 #ifndef IPU_ISYS_ISA_FW_H
 #define IPU_ISYS_ISA_FW_H
 
+#ifdef __KERNEL__
+#include <linux/types.h>
+#else
+#include <stdint.h>
+#endif
+
 #define ia_css_terminal_offsets(pg)			\
 	((uint16_t *)((void *)(pg) +			\
 		      (pg)->terminals_offset_offset))
diff --git a/include/uapi/linux/ipu-psys.h b/include/uapi/linux/ipu-psys.h
index 0f26dbfe436e..c20da184cd77 100644
--- a/include/uapi/linux/ipu-psys.h
+++ b/include/uapi/linux/ipu-psys.h
@@ -4,7 +4,11 @@
 #ifndef _UAPI_IPU_PSYS_H
 #define _UAPI_IPU_PSYS_H
 
+#ifdef __KERNEL__
 #include <linux/types.h>
+#else
+#include <stdint.h>
+#endif
 
 struct ipu_psys_capability {
 	uint32_t version;
-- 
2.20.1

